tools/system_libs.py

```diff
@@ -615,6 +615,42 @@ def get_default_variation(cls, **kwargs):
     return super().get_default_variation(eh_mode=eh_mode, **kwargs)
 
 
+class SjLjLibrary(Library):
+  def __init__(self, **kwargs):
+    # Whether we use Wasm EH instructions for SjLj support
+    self.is_wasm = kwargs.pop('is_wasm')
+    super().__init__(**kwargs)
+
+  def get_cflags(self):
+    cflags = super().get_cflags()
+    if self.is_wasm:
+      # DISABLE_EXCEPTION_THROWING=1 is the default, which is for Emscripten
+      # EH/SjLj, so we should reverse it.
+      cflags += ['-s', 'SUPPORT_LONGJMP=wasm',
+                 '-s', 'DISABLE_EXCEPTION_THROWING=1',
```

Comment:
Same question as above, I guess - not sure why we need to adjust the throwing flag. The comment here also is a little confusing for me: what is reversed?

Reply:
Oh wait, I think my confusion was that catching is disabled by default but not throwing. But then, why do we need to disable throwing? Is that a temporary workaround for something, or fundamental?

Reply:
Hmm, yes, maybe it doesn't matter.

Could we just do `-fno-exceptions` here? Or would that interfere with wasm exceptions in the future?

Anyhow, I'm ok with any of the options. Banning all exceptions for now seems least ambiguous though, if we can do that one.